etcd/v1alpha1/types_pacemakerstatus.go

```diff
@@ -0,0 +1,312 @@
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/2544
+type PacemakerStatus struct {
+	metav1.TypeMeta   `json:",inline"`
+
+	// metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec is an empty spec to satisfy Kubernetes API conventions.
+	// PacemakerStatus is a status-only resource and does not use spec for configuration.
+	// +optional
+	Spec *PacemakerStatusSpec `json:"spec,omitempty"`
+
+	// status contains the actual pacemaker cluster status information collected from the cluster.
+	// +optional
+	Status *PacemakerStatusStatus `json:"status,omitempty"`
+}
+
+// PacemakerStatusSpec is an empty spec as PacemakerStatus is a status-only resource
+type PacemakerStatusSpec struct {
+}
+
+// PacemakerStatusStatus contains the actual pacemaker cluster status information
+type PacemakerStatusStatus struct {
+	// lastUpdated is the timestamp when this status was last updated
+	// +optional
+	LastUpdated metav1.Time `json:"lastUpdated,omitempty"`
+
+	// rawXML contains the raw XML output from pcs status xml command.
+	// Kept for debugging purposes only; healthcheck should not need to parse this.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=262144
+	// +optional
+	RawXML string `json:"rawXML,omitempty"`
+
+	// collectionError contains any error encountered while collecting status
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	// +optional
+	CollectionError string `json:"collectionError,omitempty"`
+
+	// summary provides high-level counts and flags for the cluster state
+	// +optional
+	Summary *PacemakerSummary `json:"summary,omitempty"`
+
+	// nodes provides detailed information about each node in the cluster
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=2
+	// +kubebuilder:validation:MaxItems=2
+	// +optional
+	Nodes []NodeStatus `json:"nodes,omitempty"`
+
+	// resources provides detailed information about each resource in the cluster
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	Resources []ResourceStatus `json:"resources,omitempty"`
+
+	// nodeHistory provides recent operation history for troubleshooting
+	// +listType=atomic
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	NodeHistory []NodeHistoryEntry `json:"nodeHistory,omitempty"`
+
+	// fencingHistory provides recent fencing events
+	// +listType=atomic
+ 	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=16
+	// +optional
+	FencingHistory []FencingEvent `json:"fencingHistory,omitempty"`
+}
+
+// PacemakerSummary provides a high-level summary of cluster state
+type PacemakerSummary struct {
+	// pacemakerdState indicates if pacemaker is running
```

Comment:
Is the `Pacemakerd` referencing to a daemon? If so, we should probably name it `PacemakerDaemonState` to have a clearer naming.